tests/phpunit/tests/menu/wp-nav-menu.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * @group menu
+ */
+class Test_Wp_Nav_Menu extends WP_UnitTestCase {
+
+	protected $menu_id        = 0;
+	protected $lvl0_menu_item = 0;
+	protected $lvl1_menu_item = 0;
+	protected $lvl2_menu_item = 0;
+
+	public function set_up() {
+		parent::set_up();
+
+		// Create nav menu.
+		$this->menu_id = wp_create_nav_menu( 'test' );
+
+		// Create lvl0 menu item.
+		$this->lvl0_menu_item = wp_update_nav_menu_item(
+			$this->menu_id,
+			0,
+			array(
+				'menu-item-title'  => 'Root menu item',
+				'menu-item-url'    => '#',
+				'menu-item-status' => 'publish',
+			)
+		);
+
+		// Create lvl1 menu item.
+		$this->lvl1_menu_item = wp_update_nav_menu_item(
+			$this->menu_id,
+			0,
+			array(
+				'menu-item-title'     => 'Lvl1 menu item',
+				'menu-item-url'       => '#',
+				'menu-item-parent-id' => $this->lvl0_menu_item,
+				'menu-item-status'    => 'publish',
+			)
+		);
+
+		// Create lvl2 menu item.
+		$this->lvl2_menu_item = wp_update_nav_menu_item(
+			$this->menu_id,
+			0,
+			array(
+				'menu-item-title'     => 'Lvl2 menu item',
+				'menu-item-url'       => '#',
+				'menu-item-parent-id' => $this->lvl1_menu_item,
+				'menu-item-status'    => 'publish',
+			)
+		);
+	}
+
+	public function tear_down() {
+		wp_delete_nav_menu( $this->menu_id );
+		parent::tear_down();
+	}
+
+	/**
+	 * Test all menu items containing children have the CSS class `menu-item-has-children` when displaying the menu
+	 * without specifying a custom depth.
+	 *
+	 * @ticket 28620
+	 */
+	public function test_wp_nav_menu_all_depth() {
+
+		// Render the menu with all its hierarchy.
+		$menu_html = wp_nav_menu(
+			array(
+				'menu' => $this->menu_id,
+				'echo' => false,
+			)
+		);
+
+		// Level 0 should be present in the HTML output and have the `menu-item-has-children` class.
+		$this->assertStringContainsString(
```

Comment:
Test methods with multiple assertions need a `$message` value as the last argument in the assertion to explain what went wrong.

Luckily, the inline comments above each assertion will work nicely for the `$message` value.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/3357/commits/c43cda84e377cc9c95f0e07af7a4e1cbcaf3adac